### PR TITLE
Shadow Dom apis

### PR DESCRIPTION
**Pull request by bung87**
maybe there's better name for proc parameter's type

### CODE REVIEW

File: lib/js/dom.nim [hunk 1]
```diff
@@ -1503,6 +1523,19 @@ proc contains*(n: Node): bool
 proc isEqualNode*(n: Node): bool
 proc isSameNode*(n: Node): bool
 
+since (1, 3):
+  proc getRootNode*(n: Node,options: RootNodeOptions): Node
+
+  # DocumentOrShadowRoot
+  proc getSelection*(n: DocumentOrShadowRoot): Selection
+  proc elementFromPoint*(n: DocumentOrShadowRoot; x,y: float): Element
```
Comment (ringabout) [inline lib/js/dom.nim]: Need a space between `x` and `y`.

File: lib/js/dom.nim [hunk 2]
```diff
@@ -1503,6 +1523,19 @@ proc contains*(n: Node): bool
...
+  proc elementsFromPoint*(n: DocumentOrShadowRoot; x,y: float): seq[Element]
```
Comment (ringabout) [inline lib/js/dom.nim]: Need a space between x and y too.

**Comment by ringabout** [comment]
related issue:
https://github.com/nim-lang/Nim/issues/9535

**Comment by Araq** [comment]
From https://developer.mozilla.org/en-US/docs/Web/API/DocumentOrShadowRoot/elementsFromPoint

> This is an experimental technology
> Check the Browser compatibility table carefully before using this in production.

This should go into a dom_extensions.nim module instead.

**Comment by Araq** [comment]
Huh? Was that the only API that is "experimental technology"?

**Comment by bung87** [comment]
that one all browsers supported, others non-standard or draft stage

File: lib/js/dom_extensions.nim [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import dom
+
+{.push importcpp.}
```
Comment (juancarlospaco) [inline lib/js/dom_extensions.nim:3]: `{.importjs.}` ?. I think it works Ok now.

Comment (bung87) [reply lib/js/dom_extensions.nim:3]: maybe , as it is a new file, maybe ok to switch to `importjs`

**Pull request merged**